src/pytorch_lightning/trainer/trainer.py

```diff
@@ -455,9 +455,22 @@ def __init__(
         self._signal_connector = SignalConnector(self)
         self.tuner = Tuner(self)
 
-        min_steps, max_steps, min_epochs, max_epochs, max_time = _parse_loop_limits(
-            min_steps, max_steps, min_epochs, max_epochs, max_time
+        # Declare attributes to be set in _callback_connector on_trainer_init
+        self._callback_connector.on_trainer_init(
```

Comment:
I'm not happy that we are changing the order of these. We still have the issue that some of these connectors write back attributes using their internal trainer reference, and the order in which things get initialized here can matter. Have we carefully checked this? 

It looks like all you wanted to do is move `_parse_loop_limit`. Would we be able to move the callback connector and init_debugging_flags and logger connector, back to where they were? For sanity? Or have we done this to resolve another bug?

Reply:
Tests should be failing if they aren't moved (I was who initially moved it!). It's very tricky to see these dependencies between connector, the easiest way to double-check would be to move it back and run CI

Reply:
tried moving back and it failed

Reply:
I reverted the re-ordering in [64e4506](https://github.com/Lightning-AI/lightning/pull/13262/commits/64e450663f4e891f2b7f6b3971ad105eef51efe4) and all tests pass. I am still convinced not touching the ordering here is the best way for sanity.

If you still think we should be doing the re-ordering, please feel free revert my last commit, then merge the PR.